integrated_channels/moodle/models.py

```diff
@@ -64,6 +66,38 @@ class MoodleEnterpriseCustomerConfiguration(EnterpriseCustomerPluginConfiguratio
         )
     )
 
+    decrypted_username = EncryptedCharField(
+        max_length=255,
+        verbose_name="Encrypted Webservice Username",
+        blank=True,
+        help_text=_(
+            "The encrypted API user's username used to obtain new tokens."),
+        default="",
+    )
+
+    @property
+    def encrypted_username(self):
+        """
+        Return encrypted username as a string.
+
+        The data is encrypted in the DB at rest, but is unencrypted in the app when retrieved through the
+        decrypted_username field. This method will encrypt the username again before sending.
+        """
+        if self.decrypted_username:
+            return force_str(
+                self._meta.get_field('decrypted_username').fernet.encrypt(
+                    force_bytes(self.decrypted_username)
+                )
+            )
+        return self.decrypted_username
+
+    @encrypted_username.setter
+    def encrypted_username(self, value):
+        """
+        Set the encrypted username.
+        """
+        self.decrypted_username = value
+
```

Comment:
[curious] Are these properties and setters even needed? We don't need to push encrypted data anywhere in the app. Adding this property would have made sense if we were required to refer to this data in a serializer to display it somewhere in the application. Curious to know your thoughts.

Reply:
@sameenfatima78 as originally the plan was to also make changes in the code to handle encrypted fields these changes were made and kept in the same pr so changes to model are pushed here.